lib/features/thread/domain/usecases/get_all_email_interactor.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:core/core.dart';
+import 'package:dartz/dartz.dart';
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/filter/filter.dart';
+import 'package:jmap_dart_client/jmap/core/properties/properties.dart';
+import 'package:jmap_dart_client/jmap/core/sort/comparator.dart';
+import 'package:tmail_ui_user/features/thread/domain/repository/thread_repository.dart';
+import 'package:tmail_ui_user/features/thread/domain/state/get_all_email_state.dart';
+import 'package:tmail_ui_user/features/thread/domain/extensions/email_extension.dart';
+
+class GetAllEmailInteractor {
```

Comment:
`GetEmailsInMailboxInteractor`. Because we can not get all emails in one request.